File: server/router/network.go
package router

import (
	"github.com/gin-gonic/gin"

	"NanoKVM-Server/middleware"
	"NanoKVM-Server/service/network"
)

func networkRouter(r *gin.Engine) {
	service := network.NewService()

	r.POST("/api/network/wifi", service.ConnectWifiNoAuth) // connect Wi-Fi (only available in Wi-Fi configuration mode)

	api := r.Group("/api").Use(middleware.CheckToken())

	api.POST("/network/wol", service.WakeOnLAN)           // wake on lan
	api.GET("/network/wol/mac", service.GetMac)           // get mac list
	api.DELETE("/network/wol/mac", service.DeleteMac)     // delete mac
	api.POST("/network/wol/mac/name", service.SetMacName) // set mac name

	api.GET("/network/static-ip", service.GetStaticIP)  // get static IP
	api.POST("/network/static-ip", service.SetStaticIP) // set static IP

	api.GET("/network/wifi/scan", service.ScanWifi)              // scan Wi-Fi
	api.GET("/network/wifi", service.GetWifi)                    // get connected Wi-Fi
	api.POST("/network/wifi/connect", service.ConnectWifi)       // connect Wi-Fi
	api.POST("/network/wifi/disconnect", service.DisconnectWifi) // disconnect Wi-Fi
}

File: web/src/api/network.ts
import { http } from '@/lib/http.ts';

// wake on lan
export function wol(mac: string) {
  const data = {
    mac
  };
  return http.post('/api/network/wol', data);
}

// get wake-on-lan macs history
export function getWolMacs() {
  return http.get('/api/network/wol/mac');
}

export function deleteWolMac(mac: string) {
  const data = {
    mac
  };
  return http.delete('/api/network/wol/mac', data);
}

// set Mac name
export function setWolMacName(mac: string, name: string) {
  return http.post('/api/network/wol/mac/name', { mac, name });
}

// get static IP
export function getStaticIp() {
  return http.get('/api/network/static-ip');
}

// set static IP
export function setStaticIP(enabled: boolean, ip: string) {
  const data = {
    enabled: enabled,
    ip: ip
  };
  return http.post('/api/network/static-ip', data);
}

// scan wifi
export function scanWiFi() {
  return http.get('/api/network/wifi/scan');
}

// connect wifi (only available in wifi configuration mode)
export function connectWifiNoAuth(ssid: string, password: string) {
  const data = {
    ssid,
    password
  };
  return http.post('/api/network/wifi', data);
}

// connect wifi
export function connectWifi(ssid: string, password: string) {
  const data = {
    ssid,
    password
  };
  return http.post('/api/network/wifi/connect', data);
}

// disconnect wifi
export function disconnectWifi() {
  return http.post('/api/network/wifi/disconnect');
}

// get connected wifi information
export function getWiFi() {
  return http.get('/api/network/wifi');
}
